### DISCUSSION TITLE
Version of Scaffold Eth 2 that support multiple chains

### DISCUSSION THREAD

**Discussion by Ifechukwudaniel** [scaffold-eth/scaffold-eth-2#429] [Ideas]
This is a suggestion that we need ne a version of the scaffold  that supports multiple evm chains,  This core idea is to make a version of scaffold eth 2 that  will  allow developers to deploy  apps to numerous chain

**Comment by carletex** [comment]
I'd be up for this!

We'd need to tweak a few things. Off the top of my head:

1. `scaffoldConfig.targetNetwork` config an array instead of a single value.
2. Tweak `wagmiConnectors` to allow multiple target networks
3. Tweak all the logic in all the different components that use `getTargetNetwork`
4. `RainbowKitCustomConnectButton` Show a network switcher when multiple networks are configured in the app

Feel free to create an issue for it, and happy if you want to work on it ;)

Thanks @Ifechukwudaniel

**Reply by wbt** [reply]
https://github.com/scaffold-eth/scaffold-eth-2/issues/599

**Comment by ck256-2000** [comment]
good idea - multiple chains - multiple tokens - I'd be a bit game for it!

**Comment by tantodefi** [comment]
I'm working on integrating the [zeta-template](https://github.com/zeta-chain/template) repo into scaffold-eth-2 but im not exactly sure how to.
https://github.com/tantodefi/scaffold-eth-2/tree/zeta-scaffold-eth2
I thought this would be a good btc-hackathon project to better learn scaffold-eth-2
I managed to switch the network and connect to a wallet and see a aZeta balance.
in the zeta-template repo burner wallets are created with the `npx hardhat account` and they are saved to .env with the `--save` flag
but when added the @zetachain/toolkit package as a dev dependency to se2 that command prompts if i want to initiate a new hardhat project repo
i guess i need access to  @zetachain/toolkit/dist/tasks/account.js + account.d.js but somehow add it to the existing harhat scripts for command `yarn generate`
anyone have an idea how to go about doing this the correct way?
I appreciate the help speedrunnin'
